pkg/cluster/k8sres.go

```diff
@@ -735,6 +736,10 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*v1beta1.State
 		return nil, fmt.Errorf("could not generate resource requirements: %v", err)
 	}
 
+	if c.OpConfig.EnableInitContainers {
```

Comment:
The situation where the spec for init containers is provided, but the option itself is disabled, is worth logging a warning.